packages/requests/src/lib/requests-result.spec.ts

```diff
@@ -310,4 +310,95 @@ describe('requests result', () => {
 
     expect(reqSpy).toHaveBeenCalledTimes(3);
   });
+
+  it('should return loading state when staleTime passes and alwaysReturnCachedResult true', () => {
+    jest.useFakeTimers();
+
+    const { state, config } = createState(withEntities<Todo>());
+
+    const store = new Store({ state, config, name: 'todos' });
+    const reqSpy = jest.fn();
+    const entities$ = store.pipe(
+      selectAllEntities(),
+      joinRequestResult(['todos'])
+    );
+
+    const spy = jest.fn();
+
+    let i = 0;
+    function getTodos() {
+      return timer(1000).pipe(
+        switchMap(() =>
+          defer(() => {
+            i++;
+            return of([createTodo(i)]);
+          })
+        ),
+        tap((todos) => {
+          reqSpy();
+          store.update(setEntities(todos));
+        }),
+        trackRequestResult(['todos'], {
+          staleTime: 5000,
+          alwaysReturnCachedResult: true,
+        })
+      );
+    }
+
+    getTodos().subscribe();
+
+    jest.runAllTimers();
+    expect(reqSpy).toHaveBeenCalledTimes(1);
+
+    getTodos().subscribe();
+    expect(reqSpy).toHaveBeenCalledTimes(1);
+
+    jest.advanceTimersByTime(6000);
+    getTodos().subscribe();
+
+    entities$.subscribe((value) => {
```

Comment:
Please put the store subscription at the top and tell me whether it called twice when you call runAllTimers.